app/permission/foreground/main.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+/*
+Package foreground implements permissions to run a foreground service.
+See https://developer.android.com/guide/components/foreground-services.
+
+The following entries will be added to AndroidManifest.xml:
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
+*/
+
+package foreground
+
+// Start notifies the system that the program will perform background work and
+// that it shouldn't be killed. It returns a CancelFunc that should be called
+// when the background work is complete.
+
+// Start is a no-op on Linux, Windows, macOS; Android will
+// display a notification during background work; iOS isn't supported.
+func Start(title, text string) (func(), error) {
```

Comment:
Ok, I moved code out of the app/permission path and added os_nandroid.go to add a no-op Start() method for platforms other than android so that calls to Start() and its returned stop method have no effect.